src/CompatibilityFix/MovedClasses.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace AmpProject\CompatibilityFix;
+
+use AmpProject\CompatibilityFix;
+
+/**
+ * Backwards compatibility fix for classes that were moved.
+ *
+ * @package ampproject/amp-toolbox
+ */
+final class MovedClasses implements CompatibilityFix
+{
+
+    /**
+     * Mapping of aliases to be registered.
+     *
+     * @var array<string, string> Associative array of class alias mappings.
+     */
+    const ALIASES = [
+        // v0.9.0 - moved HTML-based utility into a separate `Html` sub-namespace.
+        'AmpProject\AtRule'             => 'AmpProject\Html\AtRule',
+        'AmpProject\Attribute'          => 'AmpProject\Html\Attribute',
+        'AmpProject\LengthUnit'         => 'AmpProject\Html\LengthUnit',
+        'AmpProject\RequestDestination' => 'AmpProject\Html\RequestDestination',
+        'AmpProject\Role'               => 'AmpProject\Html\Role',
+        'AmpProject\Tag'                => 'AmpProject\Html\Tag',
+
+        // v0.9.0 - extracted `Encoding` out of `Dom\Document`, as it is turned into AMP value object.
+        'AmpProject\Dom\Document\Encoding' => 'AmpProject\Encoding',
+
+    ];
+
+    /**
+     * Register the compatibility fix.
+     *
+     * @return void
+     */
+    public static function register()
+    {
+        spl_autoload_register(__CLASS__ . '::autoloader');
+    }
+
+    /**
+     * Autoloader to register.
+     *
+     * @param string $oldClassName Old class name that was requested to be autoloaded.
+     * @return void
+     */
+    public static function autoloader($oldClassName)
+    {
+        if (! array_key_exists($oldClassName, self::ALIASES)) {
+            return;
+        }
+
+        class_alias(self::ALIASES[$oldClassName], $oldClassName, true);
```

Comment:
Using the `$autoload` argument is better than checking if the `class_exists()` or `interface_exists()`?

Reply:
The `$autoload` argument is already `true` by default, I just opted to make this explicitly visible.

I'm not sure what you mean by replacing this with `class_exists()` or `interface_exists()`. If the class/interface does not yet exist and you don't autoload it here, things will just break. If it does already exist, the autoload is not being triggered.

Reply:
Ah, right.